components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -4,137 +4,96 @@
-
-    const onResults = (results: SearchResult[]) => {
-        if (JSON.stringify(results) !== JSON.stringify(searchResults)) {
-            setSearchResults(results);
-            setSelectedSearchResult(results[0]);
-        }
-    };
-
-    const search = async (query: string) => {
-        setSearchQuery(query);
-        await findResults(query, onResults);
-        if (await refreshSearchData(query, user)) {
-            // Re-run search if the underlying search data has changed
-            await findResults(query, onResults);
-        }
-    };
-
-    useEffect(() => {
-        search("");
-    }, []);
 
-    // Up/Down keyboard navigation between results
-    const onKeyDown = (event: React.KeyboardEvent) => {
-        if (!selectedSearchResult) {
-            return;
-        }
-        const selectedIndex = searchResults.indexOf(selectedSearchResult);
-        const select = (index: number) => {
-            // Implement a true modulus in order to "wrap around" (e.g. `select(-1)` should select the last result)
-            // Source: https://stackoverflow.com/a/4467559/3461173
-            const n = Math.min(searchResults.length, MAX_DISPLAYED_ITEMS);
-            setSelectedSearchResult(searchResults[((index % n) + n) % n]);
-        };
-        if (event.key === "ArrowDown") {
-            event.preventDefault();
-            select(selectedIndex + 1);
-            return;
-        }
-        if (event.key === "ArrowUp") {
-            event.preventDefault();
-            select(selectedIndex - 1);
-            return;
-        }
-    };
+interface RepositoryFinderProps {
+    initialValue?: string;
+    maxDisplayItems?: number;
+    setSelection: (selection: string) => void;
+}
 
+export default function RepositoryFinder(props: RepositoryFinderProps) {
+    const [suggestedContextURLs, setSuggestedContextURLs] = useState<string[]>(loadSearchData());
     useEffect(() => {
-        const element = document.querySelector(`a[href='/#${selectedSearchResult}']`);
-        if (element) {
-            element.scrollIntoView({ behavior: "smooth", block: "nearest" });
-        }
-    }, [selectedSearchResult]);
-
-    const onSubmit = (event: React.FormEvent) => {
-        event.preventDefault();
-        if (selectedSearchResult) {
-            window.location.href = "/#" + selectedSearchResult;
-        }
-    };
+        getGitpodService()
+            .server.getSuggestedContextURLs()
+            .then((urls) => {
+                setSuggestedContextURLs(urls);
+                saveSearchData(urls);
+            });
+    }, [suggestedContextURLs]);
+
+    const getElements = useMemo(
+        () => (searchString: string) => {
+            const result = [...suggestedContextURLs];
+            try {
+                // If the searchString is a URL, and it's not present in the proposed results, "artificially" add it here.
+                new URL(searchString);
+                if (!result.includes(searchString)) {
```

Comment:
This works when I'm pasting it in, but if I keep my url I entered selected, collapse, and open it up again, I don't seen an entry for it anymore. It is nice that it doesn't wipe out the manual URL I entered though, but was confusing to not see it as an option. Could we also look at the selected value, and if it's not in the list of options, add one for it?